src/coreclr/src/gc/gc.cpp

```diff
@@ -34111,7 +34206,11 @@ void gc_heap::sweep_uoh_objects (int gen_num)
                 {
                     dprintf (3, ("Trimming seg to %Ix[", (size_t)plug_end));
                     heap_segment_allocated (seg) = plug_end;
-                    decommit_heap_segment_pages (seg, 0);
+                    // Note the block above - it is possible for a segment to 
+                    // leave the segment chain without being decommitted in the 
+                    // previous iteration, therefore we have to skip verification
+                    // here.
+                    decommit_heap_segment_pages (seg, 0 VERIFY_COMMITTED_BY_OH_ARG (true));
```

Comment:
instead of "the block above" I would say what it actually is - it's when we delete a segment and thread it into the freeable_uoh_segment list. of course we don't *have to* skip verification - it's a choice. you can still verify it by going through the freeable_uoh_segment list.